### ISSUE TITLE
Bug: Fluent-bit doesn't send ping-message to FluentD when DB is configured for the systemd-input plugin.

### ISSUE THREAD

**Issue by miono** [fluent/fluent-bit#1817]
## Bug Report

**Describe the bug**
When enabling the `DB`-directive in conjunction with a `shared_key` for the systemd-input, logs don't appear in FluentD.
Problem exists in both v.1.3.3 and v.1.3.4.

**To Reproduce**

- Steps to reproduce the problem:

* Install td-agent-bit v1.3.4 from https://packages.fluentbit.io/ubuntu/bionic
* Install td-agent v3.5.1-0 from http://packages.treasuredata.com/3/ubuntu/bionic/

* Apply this td-agent-bit configuration:
```
[SERVICE]
    Flush         1
    Daemon        False
    Log_level     info
    Parsers_File  /etc/td-agent-bit/parsers.conf
    Mem_Buf_Limit 10M

[INPUT]
    Name          systemd
    DB            /tmp/cursor.db
    Tag           logtest

[OUTPUT]
    Name        forward
    Match       *
    Host        localhost
    Port        24224
    Shared_Key  abcd1234
```

* Apply this td-agent configuration:
```
<system>
  log_level info
</system>
<source>
  @type forward
  port 24224
  bind 0.0.0.0
  <security>
    self_hostname localhost
    shared_key abcd1234
  </security>
</source>
<match **>
  @type copy
  <store>
    @type stdout
  </store>
</match>
```
* Start td-agent-bit `sudo systemctl start td-agent-bit`
* Start td-agent without daemonizing it: `/usr/sbin/td-agent`
* Do something that will generate logs: `sudo ls`

When fluent-bit is configured with DB **enabled** it doesn't send the ping-message it's supposed to, instead it just sends the data immediately. This makes the fluentd-instance respond with "PONG invalid ping message".
When **disabling** the DB-configuration it starts out with sending the PING-message and then gets the PONG from fluentd and proceeds with sending data after that.

**Expected behavior**
td-agent should output the logs on stdout

**Screenshots**
Screenshots from Wireshark

The TCP-stream:
![TCP-stream](https://user-images.githubusercontent.com/856633/70906438-ad715600-2006-11ea-8e01-ea81405c7e0a.png)

The invalid ping message packet from FluentD:
![Invalid-ping-message](https://user-images.githubusercontent.com/856633/70906446-b19d7380-2006-11ea-831b-53290c6a1ace.png)

**Your Environment**
* Version used: Fluent-bit 1.3.4. Fluentd 3.5.1-0
* Configuration: Stated above.
* Environment name and version (e.g. Kubernetes? What version?): Running locally on my laptop.
* Server type and version: My laptop, Lenovo T480s
* Operating System and version: Ubuntu 18.04.3
* Filters and plugins: Nothing extra installed, those enabled that are in the configuration above.

**Comment by miono** [comment]
I believe this will be fixed by @bluebike s PR https://github.com/fluent/fluent-bit/pull/1783
When building from source I wasn't able to reproduce until I enabled the build-flag
-DFLB_JEMALLOC=true (Since td-agent-bit from packages.fluentbit.io seems to be built with jemalloc).

I added logging here: https://github.com/fluent/fluent-bit/blob/d73ff32a3c6e499a15168abc62e0d26c826bdc86/plugins/out_forward/forward.c#L1025 Just before the if-clause like this `flb_info("fc->shared_key is %s u_conn->ka_count is %i", fc->shared_key, u_conn->ka_count);` And got this log:
`[2019/12/16 14:37:00] [ info] fc->shared_key is abcd1234 u_conn->ka_count is 909654117`

Of course then not hitting the `secure_forward_handshake` that sends the PING-packet.

**Comment by edsiper** [comment]
Fix to be shipped as part of v1.3.5 release
